libqtile/widget/openweather.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding:utf-8 -*-
+# Copyright (c) 2020 Himanshu Chauhan
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import time
+from urllib.parse import urlencode
+
+from libqtile.widget import base
+from libqtile.widget.generic_poll_text import GenPollUrl
+
+# API access details at https://openweathermap.org/current
+QUERY_URL = "http://api.openweathermap.org/data/2.5/weather?"
```

Comment:
Looks like it's used only once, so why making it a constant? I'd rather have this as a widget option or hard-coded.

Reply:
Doesn't make sense as an option. Hardcoding directly in function can be done. This is mostly for beautification so that that lines don't go beyond 80/90 char limit. Easier on eyes.

Reply:
Again, the YahooWeather-widget does the same thing. I guess it could make sense to have this if a common base class would be extracted or anything like that. An option to the widget might be a good idea, too, since the URL could change in the future.